### PR TITLE
Support multiple events in on, off and once

### LINKED ISSUE #1: Support multiple events in `on()`
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/emittery/issues/1)

I considered adding support for multiple events in `on()`, but wanted to wait to see if this is something people actually need.

Relevant Node.js issue: https://github.com/nodejs/node/issues/15697

// @silverwind

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $40.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)

### Submitted pull Requests
- [#36 Support multiple events in on, off and once](https://issuehunt.io/r/sindresorhus/emittery/pull/36)
- [#52 support multiple events in on, off and once](https://issuehunt.io/r/sindresorhus/emittery/pull/52)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/emittery/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by Angelmmiguel**
Add support for multiple events in `on`, `off` and `once`. I started the implementation adding the `onMulti` method. However, I realized that the implementation with `string|string[]` is pretty clear and compatible with the current syntax of the methods. For that reason, I've implemented the PR in this way but feel free to ask me to define the new other methods.

**Changes**
* Support multiple events in `on`, `off` and `once`
* Add tests
* Improve docs

It closes #1

### CODE REVIEW

**Comment by Angelmmiguel** [comment]
Hey!

Do you need any other change on this PR? :)

Thanks!

File: Emittery.d.ts [hunk 1]
```diff
@@ -9,20 +9,20 @@ declare class Emittery {
 	 * Using the same listener multiple times for the same event will result
 	 * in only one method call per emitted event.
 	 */
-	on(eventName: string, listener: (eventData?: any) => any): Emittery.UnsubscribeFn;
+	on(eventName: string|string[], listener: (eventData?: any) => any): Emittery.UnsubscribeFn;
```
Comment (sindresorhus) [inline Emittery.d.ts]:
```suggestion
	on(eventName: string | string[], listener: (eventData?: any) => any): Emittery.UnsubscribeFn;
```

File: index.js [hunk 2]
```diff
@@ -31,23 +31,34 @@ class Emittery {
...
 
-	on(eventName, listener) {
-		assertEventName(eventName);
+	on(eventNames, listener) {
 		assertListener(listener);
-		getListeners(this, eventName).add(listener);
-		return this.off.bind(this, eventName, listener);
+
+		const events = Array.isArray(eventNames) ? eventNames : [eventNames];
+		events.forEach(eventName => {
```
Comment (sindresorhus) [inline index.js]: Use a `for-of` loop.

File: readme.md [hunk 3]
```diff
@@ -36,40 +36,70 @@ emitter.emit('🦄', '🌈');
 
 The above only works in Node.js 8 or newer. For older Node.js versions you can use `require('emittery/legacy')`.
 
-
```
Comment (sindresorhus) [inline readme.md]: Don't do unrelated changes.

File: readme.md [hunk 4]
```diff
@@ -36,40 +36,70 @@ emitter.emit('🦄', '🌈');
...
 ## API
 
 ### emitter = new Emittery()
 
-#### on(eventName, listener)
+#### on([eventName], listener)
 
-Subscribe to an event.
+Subscribe to an event or multiple ones.
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
Subscribe to one or more events.
```

File: readme.md [hunk 5]
```diff
@@ -36,40 +36,70 @@ emitter.emit('🦄', '🌈');
 
 The above only works in Node.js 8 or newer. For older Node.js versions you can use `require('emittery/legacy')`.
 
-
 ## API
 
 ### emitter = new Emittery()
 
-#### on(eventName, listener)
+#### on([eventName], listener)
```
Comment (sindresorhus) [inline readme.md]: The `[ ]` notation is more commonly used to indicate optional parameter. You need to find a better way to document this.

File: readme.md [hunk 6]
```diff
@@ -36,40 +36,70 @@ emitter.emit('🦄', '🌈');
...
 Using the same listener multiple times for the same event will result in only one method call per emitted event.
 
+```js
+emitter.on('🦄').then(data => {
+	console.log(data);
+});
+
+emitter.on(['🦄', '🐶']).then(data => {
+	console.log(data);
+});
```
Comment (sindresorhus) [inline readme.md]: Use async/await syntax.

**Comment by sindresorhus** [comment]
The readme changes needs to be applied to the TypeScript definition too, including examples and descriptions.

**Comment by sindresorhus** [comment]
@Angelmmiguel Still interested in finishing this PR?

**Comment by Angelmmiguel** [comment]
Hello @sindresorhus,

Yes I'm interested in finishing it. I'll try to find some time during the week to apply your feedback.

Thanks!

**Comment by sindresorhus** [comment]
@Angelmmiguel Bump :)

**Comment by LuKks** [comment]
@Angelmmiguel you actually still interested?

**Comment by Angelmmiguel** [comment]
Hello! Sorry, I'm a bit busy right now 😅. I'll probably can work on this in the following weeks, but feel free to take my PR and finish the changes or open a new one 🙂
